polylogue/ui/__init__.py

```
@@ -109,5 +117,60 @@ def _warn_plain(self, topic: str) -> None:
         self.console.print(f"[yellow]Plain mode cannot prompt for {topic}; using defaults.")
 
 
-def create_ui(flag_plain: bool) -> UI:
-    return UI(plain=detect_plain(flag_plain))
+def create_ui(plain: bool) -> UI:
+    return UI(plain=plain)
+
+
+class _NullProgressTracker:
+    def __enter__(self):
```

Ruff (0.15.9)

[warning] Missing return type annotation for special method `__enter__`

(ANN204)